From 230350f0d331f484bd2462a3a6e84a8755e91f2b Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Thu, 28 Aug 2014 16:37:13 +0800
Subject: [PATCH 0101/1594] ENGR00329352-3 dts: imx6: ocotp register access
 needs clock

commit 14e574507ad4ef69bb385cf55b91d080a44cc396 from
git://git.freescale.com/imx/linux-2.6-imx.git

Without clock, accessing ocotp register will cause system hang.
So define the clock here in device tree for driver to use.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit fa8e45c7c9865d78480043db87eba55a301edd8f)
---
 arch/arm/boot/dts/imx6qdl.dtsi |    1 +
 arch/arm/boot/dts/imx6sl.dtsi  |    1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl.dtsi b/arch/arm/boot/dts/imx6qdl.dtsi
index 2f98098..b581794 100644
--- a/arch/arm/boot/dts/imx6qdl.dtsi
+++ b/arch/arm/boot/dts/imx6qdl.dtsi
@@ -1070,6 +1070,7 @@
 			ocotp: ocotp@021bc000 {
 				compatible = "fsl,imx6q-ocotp", "syscon";
 				reg = <0x021bc000 0x4000>;
+				clocks = <&clks IMX6QDL_CLK_IIM>;
 			};
 
 			tzasc@021d0000 { /* TZASC1 */
diff --git a/arch/arm/boot/dts/imx6sl.dtsi b/arch/arm/boot/dts/imx6sl.dtsi
index 79804c2..8eaeab8 100644
--- a/arch/arm/boot/dts/imx6sl.dtsi
+++ b/arch/arm/boot/dts/imx6sl.dtsi
@@ -828,6 +828,7 @@
 			ocotp: ocotp@021bc000 {
 				compatible = "fsl,imx6sl-ocotp", "syscon";
 				reg = <0x021bc000 0x4000>;
+				clocks = <&clks IMX6SL_CLK_OCOTP>;
 			};
 
 			audmux: audmux@021d8000 {
-- 
1.7.5.4

